ERCS/erc-7527.md

```diff
@@ -0,0 +1,516 @@
+    Asset asset;
+    bytes immutableData;
+    bytes initData;
+}
+
+/**
+ * @dev The settings of the app.
+ * @param implementation The address of the app implementation.
+ * @param immutableData The immutable data are stored in the code region of the created proxy contract of appImplementation.
+ * @param initData If init data is not empty, calls proxy contract of appImplementation with this data.
+ */
+struct AppSettings {
+    address implementation;
+    bytes immutableData;
+    bytes initData;
+}
+
+interface IERC7527Factory {
+    /**
+     * @dev Deploys a new agency and app clone and initializes both.
+     * @param agencySettings The settings of the agency.
+     * @param appSettings The settings of the app.
+     * @param data The data is additional data, it has no specified format and it is sent in call to `factory`.
+     * @return appInstance The address of the created proxy contract of appImplementation.
+     * @return agencyInstance The address of the created proxy contract of agencyImplementation.
+     */
+    function deployWrap(AgencySettings calldata agencySettings, AppSettings calldata appSettings, bytes calldata data)
+        external
+        returns (address, address);
+}
+
+```
+
+## Rationale
+
+### Prior Interfaces
+
+[ERC-5679](./eip-5679.md) proposed `IERC5679Ext721` interface for introducing a consistent way to extend [ERC-721](./eip-721.md) token standards for minting and burning. To ensure the backward compatibility, considering some contracts which do not implement `ERC721TokenReceiver`, `IERC7527App` employ `mint` function instead of `safeMint`. To ensure the safety and the uniqueness of mutual bound, the `_from` parameter of the `burn` function in `IERC5679Ext721` must be the contract address of the bounded angency. Thus, `burn` function in `IERC7527App` does not contain the `_from` parameter. 
+
+### Mutual Bound
+
+Implement contracts for `IERC7527App` and `IERC7527Agency` so that they are each other's only owner. The wrap process is to check the premium amount of the fungible token received and then mint non-fungible token in the App. Only the owner or an approver of the non-fungible token can unwrap it.
+
+### Implementation Diversity 
+
+Users can customize function and fee percentage when implement the Agency and the App interfaces.
+
+Different Agency implementations have distinct wrap, unwrap function logic, and different oracleFunction. Users can customize the currency, initial price, fee receiving address, fee rate, etc., to initialize the Agency contract. 
+
+Different App implementations cater to various use cases. Users can initialize the App contract.
+
+Factory is not required. Factory implementation is need-based. Users can deploy their own contracts by selecting different Agency implementations and different App implementations through the Factory, combining them to create various products.
+
+
+### Currency types
+
+Support ETH and [ERC-20](./eip-20.md).
```

Comment:
This isn't really explaining anything. Could you elaborate?

Reply:
We aims to explain `currency` in the system, which supports FTs. 

`currency` in `IERC7527Agency` is the address of fungible token. `Asset` can only define one type of `currency` as the fungible token in the system. `currency` supports various kinds of fungible tokens including ETH and ERC-20.